### ISSUE TITLE
Issue with mocks for multiple items in a single file

### ISSUE THREAD

**Issue by a-voronov** [SwiftKit/Cuckoo#15]
When generating multiple mocks within single file, imports are being duplicated that causes compilation errors.
I'm using slightly modificated script for this purpose.
```sh
OUTPUT_FILE="./${PROJECT_NAME}Tests/CuckooMocks/GeneratedMocks.swift"
INPUT_DIR="./${PROJECT_NAME}/Classes"

find ${INPUT_DIR} -name *.swift | xargs \
cuckoo generate \
    --runtime $CUCKOO_VERSION \
    --testable ${PROJECT_NAME} \
    --output ${OUTPUT_FILE}
```

As an alternative, you create mirror of your application's target directory tree structure inside testing target directory and generate mocks for each file separately. The only problem with this is that you have to manually add generated files and directories into Xcode project.
```sh
INPUT_DIR="./${PROJECT_NAME}/Classes"
OUTPUT_DIR="./${PROJECT_NAME}Tests/CuckooMocks"

# Recursively run through the $INPUT_DIR and find files with .swift extension
find ${INPUT_DIR} -type f -iname "*.swift" -print0 | while IFS= read -r -d $'\0' filepath; do
    # Generate mirror path inside $OUTPUT_DIR
    output=${OUTPUT_DIR}${filepath#$INPUT_DIR}
    # Modify filenames by adding 'Mocks' suffix
    output=${output%".swift"}"Mocks.swift"
    # Create mirror directory tree inside $OUTPUT_DIR
    dir=$(dirname ${output})
    mkdir -p ${dir}
    
    # Run Cuckoo script to generate mock for each file
    cuckoo generate \
        --runtime $CUCKOO_VERSION \
        --testable ${PROJECT_NAME} \
        --output ${output} \
        "$filepath"
done
```

**Comment by TadeasKriz** [comment]
IIRC you can give a directory to the `--output` argument and it should be creating files in that directory, instead of writing into a single file. However, what imports are causing compile errors? I was trying multiple same imports in a single file and did not get a compilation error.

**Comment by TadeasKriz** [comment]
Anyway to solve this, we could strip the header and parse its lines to find all unique imports. Those would then be added at the top of the GeneratedMocks.swift file.

**Comment by a-voronov** [comment]
> IIRC you can give a directory to the --output argument and it should be creating files in that directory

Awesome, thanks!
Will it overwrite files with similar names?

> However, what imports are causing compile errors? I was trying multiple same imports in a single file and did not get a compilation error.

Hmm, indeed when trying to reproduce it within the working environment everything's fine.
Seems like it's different issue with scopes not being closed or else..

<img width="1027" alt="screen shot 2016-06-07 at 20 22 14" src="https://cloud.githubusercontent.com/assets/11717236/15867829/a538d8d8-2ced-11e6-8d25-9b9dc990f888.png">

**Comment by TadeasKriz** [comment]
That's very strange, I think the problem might be somewhere else, like not closing a scope with `}`.

> Will it overwrite files with similar names?
I'm not completely sure, I only used the single file approach since, so hopefully it will work well. If not, we have more to fix :)

**Comment by TadeasKriz** [comment]
Should be fixed in 0.5.1. Please try it out! :)
